packages/design-picker/src/hooks/use-categorization.ts

```diff
@@ -67,6 +67,59 @@ export function useCategorization(
 	};
 }
 
+export function useCategorizationFromApi(
+	categoryMap: Record< string, Category >,
+	{ defaultSelection, showAllFilter, generatedDesignsFilter }: UseCategorizationOptions
+): Categorization {
+	const { __ } = useI18n();
+
+	const categories = useMemo( () => {
+		const categoryMapKeys = Object.keys( categoryMap ) || [];
+		const hasCategories = !! categoryMapKeys.length;
+
+		const result = categoryMapKeys.map( ( slug ) => ( {
+			...categoryMap[ slug ],
+			slug,
+		} ) );
+
+		if ( generatedDesignsFilter && hasCategories ) {
+			result.unshift( {
+				name: generatedDesignsFilter,
+				slug: SHOW_GENERATED_DESIGNS_SLUG,
+			} );
+		}
```

Comment:
📝 for myself: This part was removed in https://github.com/Automattic/wp-calypso/pull/75999/files#diff-42cc003d4c9fd99d589d2d96fe25f7e810bb8c398323cb6212f272818c85a3a4

<!--
We might be able to extract this part as a function like `
some functions from this hook (`useCategorizationFromApi`) and `useCategorization` in the future.

-->